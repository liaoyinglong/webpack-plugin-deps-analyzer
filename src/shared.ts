export interface Options {
  /**
   * json 文件输出路径
   */
  outDir?: string;

  /**
   * 如果有不同版本的依赖，是否输出详细信息
   */
  logMultiDeps?: boolean;
  /**
   * 是否输出总共参与编译的文件数目
   */
  logFileCount?: boolean;
}
