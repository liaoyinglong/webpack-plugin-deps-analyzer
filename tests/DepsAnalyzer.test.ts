import { test, expect, mock, describe } from "bun:test";
import { runWebpackBuild } from "./shared";
const random = mock(() => Math.random());

describe("DepsAnalyzer", () => {
  test("web3Modal", async () => {
    await runWebpackBuild("web3Modal");
  });
});
