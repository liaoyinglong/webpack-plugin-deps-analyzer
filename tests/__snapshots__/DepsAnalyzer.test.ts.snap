// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DepsAnalyzer > web3Modal 1`] = `
{
  "@adraffy/ens-normalize": [
    "1.10.1",
  ],
  "@coinbase/wallet-sdk": [
    "3.9.1",
  ],
  "@lit/reactive-element": [
    "1.6.3",
    "2.0.4",
  ],
  "@metamask/safe-event-emitter": [
    "2.0.0",
    "3.1.1",
  ],
  "@metamask/utils": [
    "5.0.2",
  ],
  "@motionone/animation": [
    "10.17.0",
  ],
  "@motionone/dom": [
    "10.17.0",
  ],
  "@motionone/easing": [
    "10.17.0",
  ],
  "@motionone/generators": [
    "10.17.0",
  ],
  "@motionone/types": [
    "10.17.0",
  ],
  "@motionone/utils": [
    "10.17.0",
  ],
  "@noble/curves": [
    "1.2.0",
  ],
  "@noble/hashes": [
    "1.3.2",
  ],
  "@stablelib/binary": [
    "1.0.1",
  ],
  "@stablelib/chacha": [
    "1.0.1",
  ],
  "@stablelib/chacha20poly1305": [
    "1.0.1",
  ],
  "@stablelib/constant-time": [
    "1.0.1",
  ],
  "@stablelib/ed25519": [
    "1.0.3",
  ],
  "@stablelib/hash": [
    "1.0.1",
  ],
  "@stablelib/hkdf": [
    "1.0.1",
  ],
  "@stablelib/hmac": [
    "1.0.1",
  ],
  "@stablelib/int": [
    "1.0.1",
  ],
  "@stablelib/poly1305": [
    "1.0.1",
  ],
  "@stablelib/random": [
    "1.0.2",
  ],
  "@stablelib/sha256": [
    "1.0.1",
  ],
  "@stablelib/sha512": [
    "1.0.1",
  ],
  "@stablelib/wipe": [
    "1.0.1",
  ],
  "@stablelib/x25519": [
    "1.0.3",
  ],
  "@walletconnect/core": [
    "2.11.1",
  ],
  "@walletconnect/environment": [
    "1.0.1",
  ],
  "@walletconnect/ethereum-provider": [
    "2.11.1",
  ],
  "@walletconnect/events": [
    "1.0.1",
  ],
  "@walletconnect/heartbeat": [
    "1.2.1",
  ],
  "@walletconnect/jsonrpc-http-connection": [
    "1.0.7",
  ],
  "@walletconnect/jsonrpc-provider": [
    "1.0.13",
  ],
  "@walletconnect/jsonrpc-types": [
    "1.0.3",
  ],
  "@walletconnect/jsonrpc-utils": [
    "1.0.8",
  ],
  "@walletconnect/jsonrpc-ws-connection": [
    "1.0.14",
  ],
  "@walletconnect/keyvaluestorage": [
    "1.1.1",
  ],
  "@walletconnect/logger": [
    "2.0.1",
  ],
  "@walletconnect/modal": [
    "2.6.2",
  ],
  "@walletconnect/modal-core": [
    "2.6.2",
  ],
  "@walletconnect/modal-ui": [
    "2.6.2",
  ],
  "@walletconnect/relay-api": [
    "1.0.9",
  ],
  "@walletconnect/relay-auth": [
    "1.0.4",
  ],
  "@walletconnect/safe-json": [
    "1.0.2",
  ],
  "@walletconnect/sign-client": [
    "2.11.1",
  ],
  "@walletconnect/time": [
    "1.0.2",
  ],
  "@walletconnect/types": [
    "2.11.1",
  ],
  "@walletconnect/universal-provider": [
    "2.11.1",
  ],
  "@walletconnect/utils": [
    "2.11.1",
  ],
  "@walletconnect/window-getters": [
    "1.0.1",
  ],
  "@walletconnect/window-metadata": [
    "1.0.1",
  ],
  "@web3modal/common": [
    "4.1.1",
  ],
  "@web3modal/core": [
    "4.1.1",
  ],
  "@web3modal/ethers": [
    "4.1.1",
  ],
  "@web3modal/polyfills": [
    "4.1.1",
  ],
  "@web3modal/scaffold": [
    "4.1.1",
  ],
  "@web3modal/scaffold-utils": [
    "4.1.1",
  ],
  "@web3modal/siwe": [
    "4.1.1",
  ],
  "@web3modal/ui": [
    "4.1.1",
  ],
  "@web3modal/wallet": [
    "4.1.1",
  ],
  "aes-js": [
    "4.0.0-beta.5",
  ],
  "async-mutex": [
    "0.2.6",
  ],
  "base64-js": [
    "1.5.1",
  ],
  "bn.js": [
    "5.2.1",
  ],
  "buffer": [
    "6.0.3",
  ],
  "clsx": [
    "1.2.1",
  ],
  "cross-fetch": [
    "3.1.8",
  ],
  "dayjs": [
    "1.11.10",
  ],
  "debug": [
    "4.3.4",
  ],
  "decode-uri-component": [
    "0.2.2",
  ],
  "destr": [
    "2.0.3",
  ],
  "detect-browser": [
    "5.3.0",
  ],
  "dijkstrajs": [
    "1.0.3",
  ],
  "encode-utf8": [
    "1.0.3",
  ],
  "eth-block-tracker": [
    "7.1.0",
  ],
  "eth-json-rpc-filters": [
    "6.0.1",
  ],
  "eth-query": [
    "2.1.2",
  ],
  "eth-rpc-errors": [
    "4.0.3",
  ],
  "ethers": [
    "6.11.1",
  ],
  "eventemitter3": [
    "5.0.1",
  ],
  "events": [
    "3.3.0",
  ],
  "fast-safe-stringify": [
    "2.1.1",
  ],
  "filter-obj": [
    "1.1.0",
  ],
  "hey-listen": [
    "1.0.8",
  ],
  "idb-keyval": [
    "6.2.1",
  ],
  "ieee754": [
    "1.2.1",
  ],
  "inherits": [
    "2.0.4",
  ],
  "isomorphic-unfetch": [
    "3.1.0",
  ],
  "json-rpc-engine": [
    "6.1.0",
  ],
  "json-rpc-random-id": [
    "1.0.1",
  ],
  "keccak": [
    "3.0.4",
  ],
  "lit": [
    "2.8.0",
    "3.1.0",
  ],
  "lit-element": [
    "3.3.3",
    "4.0.4",
  ],
  "lit-html": [
    "2.8.0",
    "3.1.2",
  ],
  "lodash.isequal": [
    "4.5.0",
  ],
  "lru-cache": [
    "6.0.0",
  ],
  "motion": [
    "10.16.2",
  ],
  "ms": [
    "2.1.2",
  ],
  "multiformats": [
    "9.9.0",
  ],
  "pify": [
    "3.0.0",
    "5.0.0",
  ],
  "pino": [
    "7.11.0",
  ],
  "preact": [
    "10.20.1",
  ],
  "preact-hooks": [
    "0.1.0",
  ],
  "proxy-compare": [
    "2.5.1",
  ],
  "qrcode": [
    "1.5.3",
  ],
  "query-string": [
    "7.1.3",
  ],
  "quick-format-unescaped": [
    "4.0.4",
  ],
  "react": [
    "18.2.0",
  ],
  "readable-stream": [
    "3.6.2",
  ],
  "safe-buffer": [
    "5.2.1",
  ],
  "semver": [
    "7.6.0",
  ],
  "sha.js": [
    "2.4.11",
  ],
  "split-on-first": [
    "1.1.0",
  ],
  "strict-uri-encode": [
    "2.0.0",
  ],
  "string_decoder": [
    "1.3.0",
  ],
  "superstruct": [
    "1.0.4",
  ],
  "tslib": [
    "1.14.1",
    "2.6.2",
  ],
  "uint8arrays": [
    "3.1.1",
  ],
  "unfetch": [
    "4.2.0",
  ],
  "unstorage": [
    "1.10.2",
  ],
  "use-sync-external-store": [
    "1.2.0",
  ],
  "util-deprecate": [
    "1.0.2",
  ],
  "valtio": [
    "1.11.2",
  ],
  "ws": [
    "7.5.9",
  ],
  "xtend": [
    "4.0.2",
  ],
  "yallist": [
    "4.0.0",
  ],
  "zod": [
    "3.22.4",
  ],
}
`;
